src/hooks/use-timer.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState, useEffect } from 'react'
+
+import { ONE_SECOND, ONE_MINUTE } from '~/constants'
+import { formatDuration } from '~/utils/helper-functions'
+
+const useTimer = (initialTime: number, onTimeEnd?: () => void) => {
+  const [time, setTime] = useState(() => (initialTime <= 0 ? 0 : initialTime))
+
+  const isTimeEnds = time <= ONE_MINUTE
```

Comment:
I would suggest you rename that because I got what is only after a second review)

Also, I think you should add that `warning phase` time as a parameter of this hook. It will make it more reusable.

```suggestion
  const isInWarningPhase  = time <= ONE_MINUTE
```

Reply:
I renamed it to 'isTimeRunningOut`, which I think is more obvious. I also added time as a parameter with a default value of `ONE_MINUTE`.